src/main/java/jek/collections/IntArrayList.java

```diff
@@ -100,22 +94,37 @@ public void add(int index, Object element) {
     @Override
     public Object remove(int index) {
 
-        if (index >= end || index < 0) {
-            throw new IndexOutOfBoundsException("ты пытаешься удалить то, чего у тебя нет!"); }
+        if (index >= end || index < 0)
+            throw new IndexOutOfBoundsException("ты пытаешься удалить то, чего у тебя нет!");
 
+        Object removed = ial[index];
 
-     return null;
+        if (index < end-1)
```

Comment:
Very nice thinking about necessity , but performance vise it's very poor. 
Copying array is very heavy weight operation. Also think how many times you increase your array during adding, and make it symmetrical for removal.